File: src/main/java/com/prototype/framework/configuration/jdbc/datasource/MultiDataSourceContextHolder.java
package com.prototype.framework.configuration.jdbc.datasource;

import java.util.Objects;

public class MultiDataSourceContextHolder {

    private static final ThreadLocal<DataSourceType> contextHolder = new ThreadLocal<>();

    public static void setDataSourceType(DataSourceType dataSourceType) {
        Objects.requireNonNull(dataSourceType);
        contextHolder.set(dataSourceType);
    }

    public static DataSourceType getDataSourceType() {
        return contextHolder.get();
    }

    public static void removeContextHolder() {
        contextHolder.remove();
    }

}

File: src/main/kotlin/com/prototype/framework/configuration/jdbc/datasource/MultiDataSourceContextHolder.kt
package com.prototype.framework.configuration.jdbc.datasource

object MultiDataSourceContextHolder {
    private val contextHolder = ThreadLocal<DataSourceType>()
    var dataSourceType: DataSourceType?
        get() = contextHolder.get()
        set(dataSourceType) {
            contextHolder.set(dataSourceType)
        }

    fun removeContextHolder() {
        contextHolder.remove()
    }
}